src/transactions/TransactionFrame.cpp

```diff
@@ -606,6 +637,10 @@ TransactionFrame::apply(LedgerDelta& delta, TransactionMetaV1& meta,
         {
             processSeqNum(app.getLedgerManager(), txDelta);
         }
+        if (cv == ValidationType::kFullyValid)
```

Comment:
I think it would be cleaner to have:
* a new `ValidationType::kInvalidPostAuth` state (if `checkSignature` returns true in `TransactionFrame::commonValid`)
* call `processSignatures` if `cv >= ValidationType::kInvalidPostAuth`
    * this removes the need for `processSignature` to "rollback" `cv` like it does now and basically follows the same pattern than `processSeqNum`

Reply:
now that I think of it (spec issue?) maybe we should not `processSeqNum` if auth is not good?
I am thinking scenario of
* tx_n: setOptions(signers, ...)
* tx_n+1: oldSigners ...
there is also a different transaction `tx_n+1: newSigners` that can be executed
in this case we take the fee, but don't mess with the sequence number to avoid invalidating a potential other transaction

`txINSUFFICIENT_BALANCE` cannot happen when applying in v10 so this seems ok.

Reply:
scratch that last one: it would allow the transaction to be replayed twice (would just need to change signers or signatures)